### PR TITLE
[SPARK-36610][PYTHON] Add `thousands` argument to `ps.read_csv`.

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes adding `thousands` argument to `ps.read_csv`.

csv_file:
```csv
name;age;job;money
Jorge;30;Developer;10,000,000
Bob;32;Developer;-2,000,000
John;29;Developer;
```

before:
```python
>>> ps.read_csv(path, sep=";").money
0    10,000,000
1    -2,000,000
2          None
Name: money, dtype: object
```

after:
```python
>>> ps.read_csv(path, sep=";", thousands=",").money
0    10000000.0
1    -2000000.0
2           NaN
Name: money, dtype: float64
```

### Why are the changes needed?

We should support the API same as pandas, as much as possible.

### Does this PR introduce _any_ user-facing change?

Yes, now the `thousands` argument is supported for `ps.read_csv`.

### How was this patch tested?

Unittest

### CODE REVIEW

File: python/pyspark/pandas/namespace.py [hunk 1]
```diff
@@ -407,6 +410,20 @@ def read_csv(
         index_spark_column_names = []
         index_names = []
 
+    data_spark_columns = [scol_for(sdf, col) for col in column_labels.values()]
+    if thousands is not None:
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/namespace.py:414]: I think you can set `locale` option in CSV. Should probably find one standard that adds `,` in numbers. e.g.) see https://docs.oracle.com/cd/E19455-01/806-0169/overview-9/index.html

Comment (itholic) [reply python/pyspark/pandas/namespace.py:414]: I think seems like pandas just simply replace the string specified by `thousands` parameter to empty string, if the column is possible to cast to numeric type, regardless of locale standard ??

For example,

```csv
name;age;job;money
Jorge;30;Developer;10000,,00,0,0,0,0
Bob;32;Developer;-1234,424,142424,0
```

```python
>>> pd.read_csv(path, sep=";", thousands=",")
    name  age        job           money
0  Jorge   30  Developer     10000000000
1    Bob   32  Developer -12344241424240
```

File: python/pyspark/pandas/namespace.py [hunk 2]
```diff
@@ -407,6 +410,20 @@ def read_csv(
...
+        if not isinstance(thousands, str) or len(thousands) != 1:
+            raise ValueError("Only length-1 comment characters supported")
+        new_data_spark_columns = list()
+        for scol in data_spark_columns:
+            scol_replaced = F.regexp_replace(scol, thousands, "")
+            cond = F.when(scol_replaced.isNull(), np.nan).otherwise(scol_replaced.cast("float"))
+            if len(sdf.select(cond).filter(cond.isNull()).head(1)) == 0:
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/namespace.py:421]: can we avoid launching jobs for every column?

File: python/pyspark/pandas/namespace.py [hunk 3]
```diff
@@ -407,6 +410,20 @@ def read_csv(
         index_spark_column_names = []
         index_names = []
 
+    data_spark_columns = [scol_for(sdf, col) for col in column_labels.values()]
+    if thousands is not None:
+        if not isinstance(thousands, str) or len(thousands) != 1:
+            raise ValueError("Only length-1 comment characters supported")
+        new_data_spark_columns = list()
+        for scol in data_spark_columns:
+            scol_replaced = F.regexp_replace(scol, thousands, "")
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/namespace.py:419]: This isn't really correct. It assumes that the string representation of other types are same as the raw data but that's not the true.

For example, it wouldn't work if you set `timestampFormat` to something like `yyyy-MM` and set `thousands` to `-`.

**Comment by itholic** [comment]
Let me close this for now, since current implementation way is too hacky.

Seems like we should first handle around the CSVInferSchema and revisit later.

**Pull request closed without merging**